src/Castle.Windsor.Extensions.DependencyInjection/SubSystems/DependencyInjectionNamingSubsystem.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2004-2020 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+namespace Castle.Windsor.Extensions.DependencyInjection.SubSystems
+{
+	using Castle.MicroKernel;
+	using Castle.MicroKernel.SubSystems.Naming;
+	using Castle.MicroKernel.Util;
+	
+	using System;
+	using System.Collections.Generic;
+	using System.Reflection;
+
+	/// <summary>
+	/// Naming subsystem based on DefaultNamingSubSystem but GetHandlers returns handlers in registration order
+	/// </summary>
+	internal class DependencyInjectionNamingSubsystem :  DefaultNamingSubSystem
```

Comment:
Adding this sub-system to an existing container wipes out all its components. Furthermore, the sub-system looks virtually identical to the base implementation.

Can someone tell me why this is needed for .NET core 3.1? Is there something different about .NET core 3.1 that requires this particular class? I've compiled my own version that doesn't include this sub-system, just to test, and it seems to work the same way regardless.

Reply:
Hi @robertcoltheart 
I tried removing the subsystem when i refactored some of this code. From what I remembered it made all or many if the unit tests fail.

If you look at the new implementation you on master you will see you can override the sub system.

Could you please make a new issue, and reference the current implementation and we can look deeper into it together?